multi/stm32l4-multi/rtc.c

```diff
@@ -232,6 +241,124 @@ int rtc_setAlarm(rtctimestamp_t *timestamp)
 }
 
 
+static int rtc_getLastStorage(uint32_t *lastID)
+{
+	uint32_t id[2], valid = 0, num;
+	int ret;
+
+	id[0] = *(rtc_common.base + BACKUP1_ID_REG);
+	id[1] = *(rtc_common.base + BACKUP2_ID_REG);
+
+	if (ID_IS_VALID(id[0])) {
+		valid |= 1;
+	}
+
+	if (ID_IS_VALID(id[1])) {
+		valid |= 2;
+	}
+
+	switch (valid) {
+		case 0: /* No valid records */
+			ret = -1;
+			num = 0;
+			break;
+
+		case 1: /* Only record #1 is valid */
+			ret = 0;
+			num = ID_VALUE(id[0]);
+			break;
+
+		case 2: /* Only record #2 is valid */
+			ret = 1;
+			num = ID_VALUE(id[1]);
+			break;
+
+		default: /* Both records are valid */
+			if (ID_VALUE(id[0]) > ID_VALUE(id[1])) {
```

Comment:
On id[0] = 0xffff and id[1] = 0x0 because of overflow  this condition is invalid. What worse - it never repair because we will always write new records to id[1]

Reply:
Indeed, good catch. I'll think about a fix